litgpt/tokenizer.py

```diff
@@ -136,3 +136,30 @@ def decode(self, tensor: torch.Tensor) -> str:
             dummy_token = self.processor.decode([dummy_token_id])
             return self.processor.decode([dummy_token_id] + tokens)[len(dummy_token) :]
         return self.processor.decode(tokens)
+
+    def decode_stream(self, token_stream: Iterator[torch.Tensor]) -> Iterator[str]:
+        if self.backend == "huggingface":
+            try:
+                for token in token_stream:
+                    yield self.decode(token)
+            except KeyboardInterrupt:
+                return
+        elif self.backend == "sentencepiece":
+            # TODO: Is there a way to not have to do this?
+            # This may actually affect our tokens per second.
+
+            # sentencepiece does not support decoding token-by-token because it adds spaces based on the surrounding tokens
```

Comment:
@Andrei-Aksionov reimplemented the tokenizer pipeline and may have ideas here

Reply:
Interesting. I think I didn't test the ~hack~ fix in `decode` method (with a dummy_token_id) for SentencePiece tokenizer.
So maybe now the logic below is not needed.